src/hostapi/alsa/pa_linux_alsa.c

```diff
@@ -2750,17 +2757,20 @@ static PaError PaAlsaStream_Configure( PaAlsaStream *self, const PaStreamParamet
         assert( self->capture.framesPerPeriod != 0 );
         PA_ENSURE( PaAlsaStreamComponent_FinishConfigure( &self->capture, hwParamsCapture, inParams, self->primeBuffers, realSr,
                     inputLatency ) );
+        /* Now that we have finalized the hwParams, we can get a more accurate sample rate. */
+        ENSURE_( GetExactSampleRate( hwParamsCapture, &realSr ), paUnanticipatedHostError );
         PA_DEBUG(( "%s: Capture period size: %lu, latency: %f\n", __FUNCTION__, self->capture.framesPerPeriod, *inputLatency ));
     }
     if( self->playback.pcm )
     {
         assert( self->playback.framesPerPeriod != 0 );
         PA_ENSURE( PaAlsaStreamComponent_FinishConfigure( &self->playback, hwParamsPlayback, outParams, self->primeBuffers, realSr,
                     outputLatency ) );
+        /* Now that we have finalized the hwParams, we can get a more accurate sample rate. */
+        ENSURE_( GetExactSampleRate( hwParamsPlayback, &realSr ), paUnanticipatedHostError );
```

Comment:
What should we do if the exact input and output sample rates are not the same?
We should log a warning if abs(diff)>tolerance.

Then use the average?